### DISCUSSION TITLE
Annotation with custom database

### DISCUSSION THREAD

**Discussion by SilasK** [metagenome-atlas/atlas#480] [Q&A]
@ Stefan-L asked how to annotate genes with custom genes. 

My response would be that obviously, it's best to check if there is anything in the EggNOG gene annotations or in the genome annotation produced by DRAM (e.g. KEgg) that corresponds to your gene of interest. It is also possible to annotate MAGs with UniRef with DARAM but this required big databases. 

So what to do if you are interested in a new custom gene annotation. 
First, you would need some genes that are known to have your function. And then you can search the genecatalog against them. Some time ago I made this small pipeline which would do this. https://github.com/SilasK/MMprofiler

I'm also interested in your ideas @jmtsuji @LeeBergstrand with metaanotate.

**Comment by jmtsuji** [comment]
@SilasK @ Stefan-L  Are you hoping to annotate genes using a large sequence database (e.g., Uniprot) or using a small collection of custom genes that are of interest to your project? This will affect the best method to use.

In the first case, I think Silas' answer is already pretty complete. EggNOG/DRAM annotations should be fairly comprehensive, and DRAM can support other databases (e.g, UniRef) if needed. There are also other bioinformatics tools out there specifically designed for annotation via other sequence databases.

In the second case, the database will vary a lot depending on the project. In some cases, if you can find some HMMs that target the genes you are interested in (e.g., using Pfam or FunGene), you can use hmmsearch to quickly scan the ATLAS Genecatalog to look for hits. You can then analyze those hits in more detail using multiple sequence alignments and such. If you want to use MMSeqs, you can use the seed sequences of the HMM to create a MMSeqs Profile for the search. In other cases, tools sometimes exist to specifically profile the type of genes you are interested in (e.g., FeGenie for iron-cycling genes), and it might be best to just use those tools directly and apply them to the ATLAS data.

MetAnnotate is mainly designed for annotating short read data, i.e., peptides predicted directly from unassembled metagenome reads. It relies on the RefSeq database for taxonomic annotation of short read hits.

Hope this helps!

**Reply by SilasK** [reply]
Thank you for your explanation. 

Ok, in this case metaanotate is something complementary to atlas not something to annotate atlas geses.. 

Yes my pipline creates mmseqs profiles from a bunch of genes..

**Reply by jmtsuji** [reply]
@SilasK Agreed, MetAnnotate is a complementary tool to the assembly-based approach used by ATLAS. That's nice that your pipeline creates MMSeqs profiles.

**Reply by LeeBergstrand** [reply]
@SilasK @jmtsuji Most of what I do nowadays takes the protein sequences from the contig-derived gene catalogue and scans them with custom builtt HMMs. I then filter down to matches using HMMER cut-off values. If I need abundance information I use the gene counts using the gene counts feature of the latest ATLAS release.

**Reply by LeeBergstrand** [reply]
So you basically have the same technique. MMSeqs profiles are PSSMs, I believe versus HMMER profiles are HMMs.

**Reply by LeeBergstrand** [reply]
MetAnnotate is mainly designed to run on fraggenescan derived protein fragments for raw metagenomic reads. I think you could theoretically, use it to annotate whole proteins as well. However, the key aspect of Metannotate is that it also taxonomically classifies your HMM hits so you get a sense of what organisms in your sample possess your gene of interest and what their abundances are (based on taxonomy classification of genes). The taxonomy classification aspect would be generally incompatible with gene catalogues since the clustered representative genes represent genes from multiple organisms.

**Reply by LeeBergstrand** [reply]
![12915_2015_195_Fig1_HTML gif](https://user-images.githubusercontent.com/5819462/144327027-360f78e3-ba16-4542-a675-0fdeebe5143a.jpg)

**Reply by SilasK** [reply]
Thank you for the clarifications
